src/classes/PREN_Affiliation_TDTM.cls

```diff
@@ -60,14 +60,16 @@ public class PREN_Affiliation_TDTM extends TDTM_Runnable {
     }
     
     private void createRelatedAffls(List<SObject> newlist, TDTM_Runnable.Action triggerAction) {
+	
         List<Affiliation__c> newAffls = new List<Affiliation__c>();
+
         for (SObject so : newlist) {
             Program_Enrollment__c enrollment = (Program_Enrollment__c)so;
             
             if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-                //Automatically create an Affiliation record if a Program Enrollment record has been manually created from a Contact.
+                //Automatically create an Affiliation record if a Program Enrollment record has been created without an affiliation.
                 if(enrollment.Contact__c != null && enrollment.Account__c != null && enrollment.Affiliation__c == null) {
-                    Affiliation__c affl = new Affiliation__c(Contact__c = enrollment.Contact__c, Account__c = enrollment.Account__c);
+                    Affiliation__c affl = new Affiliation__c(Contact__c = enrollment.Contact__c, Account__c = enrollment.Account__c, EndDate__c = (UTIL_CustomSettingsFacade.getSettings().Affl_ProgEnroll_Copy_End_Date__c == true ? enrollment.End_Date__c : null), StartDate__c = (UTIL_CustomSettingsFacade.getSettings().Affl_ProgEnroll_Copy_Start_Date__c == true ? enrollment.Start_Date__c : null), Role__c = (UTIL_CustomSettingsFacade.getSettings().Affl_ProgEnroll_Set_Role__c == true ? UTIL_CustomSettingsFacade.getSettings().Affl_ProgEnroll_Role_Map__c : null), Status__c = UTIL_CustomSettingsFacade.getSettings().Affl_ProgEnroll_Status_Map__c);
```

Comment:
Could we perhaps introduce some short lived variables here instead of using ternaries in the constructor? It would make it more readable in my opinion.